src/fontloader/runtime/fontloader-basics-gen.lua

```diff
@@ -253,7 +253,7 @@ do
     end
 
     if not writable then
-        logs.report("system","no writeable cache path, quiting")
+        logs.report("system","no writeable cache path, quiting. Try setting TEXMFCACHE or TEXMFVAR and reading https://www.tug.org/texinfohtml/kpathsea.html#Safe-filenames")
```

Comment:
(Not a `luaotfload` developer, so feel free to ignore any of my comments here)

I don't think that `logs.report` wraps its output (assuming that it behaves like the version from ConTeXt), so it might be better to add the hint to a separate line:


```suggestion
        logs.report("system","no writeable cache path, quiting")
        logs.report("system","Try setting TEXMFCACHE or TEXMFVAR and reading https://www.tug.org/texinfohtml/kpathsea.html#Safe-filenames")
```

In general though, adding a hint to this error message seems like a really good idea to me.

Reply:
I would like to be able to do this, but because of https://github.com/latex3/luaotfload/blob/ecfe7d02bd3afec0a68c5892b8d89a1ef3fc30f6/src/luaotfload-init.lua#L293-L306 only the last log is reported.

(I'm not a lua dev, so could easily be missing something here)

Trying to embed a newline in the string also makes the reporting a bit odd wrt indentation.
```
luaotfload | load : FATAL ERROR
luaotfload | load :   × Failed to load "fontloader" module "basics-gen".
luaotfload | load :   × Error message:
luaotfload | load :     × "./luaotfload-init.lua:301: system : no writeable cache path, quiting.\
Try setting TEXMFCACHE or TEXMFVAR and reading https://www.tug.org/texinfohtml/kpathsea.html#Safe-filenames".

...

./luaotfload-init.lua:301: system : no writeable cach
e path, quiting.
Try setting TEXMFCACHE or TEXMFVAR and reading https://www.tug.org/texinfohtml/
kpathsea.html#Safe-filenames
```

whereas just one long string gives
```
luaotfload | load : FATAL ERROR
luaotfload | load :   × Failed to load "fontloader" module "basics-gen".
luaotfload | load :   × Error message:
luaotfload | load :     × "./luaotfload-init.lua:301: system : no writeable cache path, quiting. Try setting TEXMFCACHE or TEXMFVAR and reading https://www.tug.org/texinfohtml/kpathsea.html#Safe-filenames".

...

./luaotfload-init.lua:301: system : no writeable cach
e path, quiting. Try setting TEXMFCACHE or TEXMFVAR and reading https://www.tug
.org/texinfohtml/kpathsea.html#Safe-filenames
```

I'm not particularly happy with either option here. Hopefully a maintainer can suggest a non-invasive change with nicer output.

Reply:
Either option is fine really. Currently the output looks weird but that shouldn't be worked around here but fixed in the output printer.

I think doing multiple calls to `texio.reporter` is the cleanest option for now.